presto-main/src/main/resources/webapp/src/components/SQLClient.jsx

```diff
@@ -275,55 +294,247 @@ function QueryResults({results}) {
+
+    const highlightAltered = [
+        {
+            when: row => row[1] !== row[2],
+            style: {
+                fontWeight: 'bold',
+                color: '#DFC7F2',
+            },
+        }
+    ];
+
+    const COLUMNS = [
+        {
+            name: 'Name',
+            selector: row => row[0],
+            sortable: true,
+            wrap: true,
+            minwidth: '350px',
+        },
+        {
+            name: 'Current Value',
+            selector: row => row[1],
+            cell: row => <EditableCell row={row} changeHandler={internalChangeHandler}/>,
+            width: '150px',
+            wrap: true,
+        },
+        {
+            name: 'Default Value',
+            selector: row => row[2],
+            width: '150px',
+            wrap: true,
+        },
+        {
+            name: 'Date Type',
+            selector: row => row[3],
+            width: '100px',
+            wrap: true,
+        },
+        {
+            name: 'Description',
+            selector: row => row[4],
+            minwidth: '100px',
+            wrap: true,
+        },
+    ];
+
+    const rowClicked = (row) => {
+        if (clickedRow.current.length > 0 && clickedRow.current[0] !== row[0]) {
+            // change the previous one back to readOnly mode
+            clickedRow.current[5] = true;
+        }
+        // update readOnly as false to trigger the useEffect function of EditableCell
+        row[5] = false;
+        clickedRow.current = row;
+        setFilter({...filter});
+    };
+
+    React.useEffect(() => {
+        //fetch session properties:
+        async function getSessionProps() {
+            const client = createClient();
+            const prestoQuery = await client.query('show session');
+            // Add an extra field to store the editing mode (readOnly=true|false)
+            sesstionProps.current = {data: prestoQuery.data.map(prop => [...prop, true])};
+            setFilter({text: '', data: sesstionProps.current.data});
+        }
+        getSessionProps();
+    }, []);
+
+    return (
+        <div className={show === true ? 'show' : 'hide'}>
+            <div className="row">
+                <div className='col-xs-12'>
+                    <DataTable
+                        columns={COLUMNS}
+                        data={filter.data}
+                        theme='dark'
+                        customStyles={CUSTOM_STYLES}
+                        subHeader
+                        onRowClicked={rowClicked}
+                        subHeaderComponent={<NameFilter filterText={filter.text} onFilter={filterData} onClear={clearFilter} />}
+                        pagination
+                        dense
+                        conditionalRowStyles={highlightAltered}
+                        highlightOnHover
+                        persistTableHead
+                    />
+                </div>
+            </div>
+        </div>
+    );
+}
+
 export default function SQLClientView() {
```

Comment:
I see that there are several components on the same file, consider breaking this into separate files to avoid having very long files (300+ lines) with more than one component. One component per file tends to be a good rule of thumb for react stuff.